docs/reference/contrib/modeladmin/migrating_to_snippets.md

```diff
@@ -0,0 +1,141 @@
+
+| `ModelAdmin` attribute | `SnippetViewSet` attribute | Notes |
+| ---------------------- | -------------------------- | ----- |
+| `add_to_admin_menu`    | {attr}`~SnippetViewSet.add_to_admin_menu`        | Same attribute name, but the value defaults to `False` instead of `True`. Set to `True` to add a top-level menu item for the model. |
+| `menu_icon`            | {attr}`~SnippetViewSet.icon`                     | Same value, but different attribute name, as the icon is used throughout the admin and not just in the menu. |
+| `list_display`         | {attr}`~SnippetViewSet.list_display`             | Same attribute name, but the list/tuple of strings must refer to existing attributes or methods on the model, not the `SnippetViewSet` class. If you have specified a string that refers to an attribute or method on the `ModelAdmin` class, you need to move it to the model. In addition, `list_display` now also supports instances of the `wagtail.admin.ui.tables.Column` component class. |
+| `list_filter`          | {attr}`~SnippetViewSet.list_filter`              | Same attribute name and value, but filtering is built on top of the django-filter package under the hood, which behaves differently to ModelAdmin's filters. See documentation for `SnippetViewSet.list_filter` and {attr}`~SnippetViewSet.filterset_class` for more details. |
+| `form_fields_exclude`  | {attr}`~wagtail.admin.viewsets.model.ModelViewSet.exclude_form_fields`           | Same value, but different attribute name to better align with `ModelViewSet`. |
+| - | {attr}`~SnippetViewSet.template_prefix` | New attribute. Set to the name of a template directory to override the `"wagtailsnippets/snippets/"` default. If set to `"modeladmin/"`, the template directory structure will be equal to what ModelAdmin uses. Make sure any custom templates are placed in the correct directory according to this prefix. See [](wagtailsnippets_templates) for more details. |
+
+## Unsupported features and customisations
+
+Some features and customisations in `ModelAdmin` are not directly supported via `SnippetViewSet`, but may be achievable via other means that are more in line with Wagtail's architecture.
+
+### Using ModelAdmin to manage Page models
+
+ModelAdmin allows the registration of Page models, but the main use case is to create custom page listing views. The create and edit views are not supported by ModelAdmin, as there are page-specific operations in those views that are best handled by Wagtail's page views. For this reason, registering a Page model as a snippet is not supported.
+
+In a future release, Wagtail will introduce a new "treeless" listing view for pages, as outlined in [RFC 082: Treeless page listings](https://github.com/wagtail/rfcs/pull/82) and the [Universal Listings discussion](https://github.com/wagtail/wagtail/discussions/10446). This feature will allow for custom page listing views and will be the recommended way to achieve this use case.
```

Comment:
I think maybe this could be a note style callout, it's not really documentation but added context for the community. It may also be good to suggest that people add their feedback to these discussions.

Reply:
Good call. I wasn't sure about including this though, so we can also remove this if desired.